.github/workflows/release.yml

```diff
@@ -30,19 +31,34 @@ jobs:
         with:
           go-version-file: 'go.mod'
           cache: false
+      # 1. Run snapshot build on everything except tags (verifies build without publishing/signing)
+      -
+        name: Run GoReleaser (snapshot build)
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+        uses: goreleaser/goreleaser-action@v6
+        with:
+          # 'latest', 'nightly', or a semver
+          version: '~> v2'
+          args: release --snapshot --skip=publish --skip=sign --clean
+
+      # 2. Import GPG key only for Tagged Releases
       -
         name: Import GPG key
         uses:  crazy-max/ghaction-import-gpg@v6
+        if: startsWith(github.ref, 'refs/tags/')
         id: import_gpg
         with:
           gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
           passphrase: ${{ secrets.PASSPHRASE }}
+
+      # 3. Publish Release only on Tagged Releases
       -
-        name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v4
+        name: Run GoReleaser (release build)
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: goreleaser/goreleaser-action@v6
         with:
           # 'latest', 'nightly', or a semver
-          version: '~> v1'
+          version: '~> v2'
           args: release --rm-dist
```

Comment:
🤔 does `rm-dist` flag still exist or is this from the old version? Looking at [flags docs](https://goreleaser.com/cmd/goreleaser_release/) I don't see it.

Reply:
good catch, thanks. afaict --rm-dist became --clean in v1.15.0 and it was removed in v2.

https://github.com/goreleaser/goreleaser/releases/tag/v1.15.0